examples/soft-kmeans/README.md

```diff
@@ -0,0 +1,108 @@
+# Soft k-means clustering
+Soft k-means, also known as fuzzy clustering, is the probabilistic version of the better knwon k-means clustering.
+In this example, we simulate a one dimension clustering problem to illustrate a use-case of Chainsail.
+
+
+
+## Data generation
+We generated two 1-dimension clusters of points by drawing them from two normal probability distributions:
+- **Cluster 1:** 80 points, drawn for a normal probability distribution of mean `mu1 = 0` and standard deviation `sigma = 1`
+- **Cluster 2:** 120 points, drawn for a normal probability distribution of mean `mu2 = 1.5` and standard deviation `sigma = 1`
+
+<img alt="data points" src="./images/data_points_1D.png" height="400"/>
+
+
+
+## Model
+The model used to fit those clusters is a gaussian mixture (inspired from the [Stan example of soft k-means](https://mc-stan.org/docs/2_21/stan-users-guide/soft-k-means.html)).
+It is defined in the file `probability.py`.  
+
+- **Number of mixture components:** Just like in regular "hard" k-means, the users defines the number of clusters to find a priori. There are two clusters to find in this case, so the mixture model was defined with 2 components.
```

Comment:
Can we label the clusters in the plots? Like, with texts saying "cluster 1" and "cluster 2"?

Reply:
There are labelled with a legend on the first plot. Then do you think we should show the underlying true clusters on the subsequent plots as well ?